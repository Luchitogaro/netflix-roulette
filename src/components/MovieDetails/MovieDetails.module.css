.movie-details {
  display: flex;
  gap: 2rem;
  align-items: flex-start;
  background: #232323;
  padding: 2rem;
  border-radius: 12px;
}

.poster {
  width: 260px;
  border-radius: 12px;
  box-shadow: 0 4px 24px rgba(66, 66, 66, 0.4);
}

.details {
  flex: 1;
  color: #ffffff;
}

.header {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.title {
  font-size: 2.2rem;
  font-weight: 700;
  margin: 0;
  color: #ffffff;
}

.rating-badge {
  background: #f65261;
  color: #ffffff;
  font-weight: bold;
  border-radius: 50%;
  padding: 0.5rem 1rem;
  font-size: 1.2rem;
  border: 2px solid #555555;
}

.genres {
  margin: 0.5rem 0 1rem 0;
}

.genre-tag {
  background: #424242;
  color: #ffffff;
  border-radius: 20px;
  padding: 0.3rem 1rem;
  margin-right: 0.5rem;
  font-size: 0.95rem;
  display: inline-block;
}

.meta {
  display: flex;
  gap: 2rem;
  font-size: 1.1rem;
  color: #f65261;
  margin-bottom: 1rem;
}

.year,
.duration {
  font-weight: 500;
  color: #f65261;
}

.description {
  font-size: 1.1rem;
  color: #555555;
  margin-top: 1rem;
}
