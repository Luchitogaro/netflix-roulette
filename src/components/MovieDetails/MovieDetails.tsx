import React from 'react';
import styles from './MovieDetails.module.css';
import { Movie } from '../../types';

interface MovieDetailsProps {
  movie: Movie;
}

export const MovieDetails: React.FC<MovieDetailsProps> = ({ movie }) => {
  return (
    <div
      className={styles['movie-details']}
      data-cy="movie-details"
      data-testid="movie-details"
    >
      <img
        className={styles.poster}
        src={movie.posterUrl}
        alt={movie.title}
        data-cy="movie-poster"
        data-testid="movie-poster"
      />
      <div
        className={styles.details}
        data-cy="movie-details-info"
        data-testid="movie-details-info"
      >
        <div className={styles.header} data-testid="movie-header">
          <h1
            className={styles.title}
            data-cy="movie-title"
            data-testid="movie-title"
          >
            {movie.title}
          </h1>
          {movie.rating && (
            <span
              className={styles['rating-badge']}
              data-cy="movie-rating"
              data-testid="movie-rating"
            >
              {movie.rating}
            </span>
          )}
        </div>
        {movie.genres && (
          <div
            className={styles.genres}
            data-cy="movie-genres"
            data-testid="movie-genres"
          >
            {movie.genres.map((genre) => (
              <span
                key={genre}
                className={styles['genre-tag']}
                data-testid={`movie-genre-${genre}`}
              >
                {genre}
              </span>
            ))}
          </div>
        )}
        <div className={styles.meta} data-testid="movie-meta">
          <span
            className={styles.year}
            data-cy="movie-year"
            data-testid="movie-year"
          >
            {movie.releaseYear}
          </span>
          <span
            className={styles.duration}
            data-cy="movie-duration"
            data-testid="movie-duration"
          >
            {movie.duration}
          </span>
        </div>
        <p
          className={styles.description}
          data-cy="movie-description"
          data-testid="movie-description"
        >
          {movie.description}
        </p>
      </div>
    </div>
  );
};
