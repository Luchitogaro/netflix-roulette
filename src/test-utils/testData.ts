import { Movie } from '../types';

// Single movie for basic testing
export const mockMovie: Movie = {
  id: '1',
  posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
  title: 'Bohemian Rhapsody',
  releaseYear: 2018,
  rating: 8.5,
  duration: '2h 35min',
  description: 'A biographical drama about the legendary British rock band Queen and their lead vocalist Freddie Mercury.',
  genres: ['Drama', 'Biography', 'Music'],
};

// Basic movie for MovieTile testing (without optional fields)
export const mockBasicMovie = {
  id: '1',
  posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
  title: 'Bohemian Rhapsody',
  releaseYear: 2018,
  genres: ['Drama', 'Biography', 'Music'],
};

// Movie for MovieDetails testing
export const mockDetailedMovie: Movie = {
  id: '1',
  posterUrl: 'https://example.com/poster.jpg',
  title: 'Pulp Fiction',
  releaseYear: 1994,
  rating: 8.9,
  duration: '2h 34min',
  description: 'A classic crime film directed by Quentin Tarantino.',
  genres: ['Crime', 'Drama'],
};

// Array of movies for testing multiple items
export const mockMovies: Movie[] = [
  {
    id: '1',
    posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
    title: 'Bohemian Rhapsody',
    releaseYear: 2018,
    rating: 8.5,
    duration: '2h 35min',
    description: 'A biographical drama about the legendary British rock band Queen and their lead vocalist Freddie Mercury.',
    genres: ['Drama', 'Biography', 'Music'],
  },
  {
    id: '2',
    posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
    title: 'The Shawshank Redemption',
    releaseYear: 1994,
    rating: 9.3,
    duration: '2h 22min',
    description: 'Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.',
    genres: ['Drama', 'Crime'],
  },
  {
    id: '3',
    posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
    title: 'Inception',
    releaseYear: 2010,
    rating: 8.8,
    duration: '2h 28min',
    description: 'A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a C.E.O.',
    genres: ['Action', 'Adventure', 'Sci-Fi'],
  },
];

// Movie for Cypress tests
export const cypressTestMovie: Movie = {
  id: '1',
  posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
  title: 'Bohemian Rhapsody',
  releaseYear: 2018,
  rating: 8.5,
  duration: '2h 35min',
  description: 'A biographical drama about the legendary British rock band Queen and their lead vocalist Freddie Mercury.',
  genres: ['Drama', 'Biography', 'Music'],
};

// Basic movie for Cypress tests
export const cypressBasicTestMovie = {
  id: '1',
  posterUrl: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
  title: 'Bohemian Rhapsody',
  releaseYear: 2018,
  genres: ['Drama', 'Biography', 'Music'],
};

// Genres for testing
export const mockGenres = ['ALL', 'COMEDY', 'HORROR', 'CRIME', 'DOCUMENTARY', 'DRAMA', 'ACTION', 'ADVENTURE', 'SCI-FI', 'BIOGRAPHY', 'MUSIC'];

// Search queries for testing
export const mockSearchQueries = ['Avengers', 'Batman', 'Spider-Man', 'Iron Man', 'Black Panther'];

// Sort options for testing
export const mockSortOptions = ['releaseDate', 'title', 'rating', 'duration'] as const;
