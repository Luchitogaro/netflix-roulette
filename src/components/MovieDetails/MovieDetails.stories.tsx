import type { Meta, StoryObj } from '@storybook/react-webpack5';
import { MovieDetails } from './MovieDetails';
import { mockMovie } from '../../test-utils/testData';

const meta: Meta<typeof MovieDetails> = {
  title: 'Components/MovieDetails',
  component: MovieDetails,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    movie: mockMovie,
  },
};
