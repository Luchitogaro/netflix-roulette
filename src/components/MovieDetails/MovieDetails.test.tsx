import React from 'react';
import { render } from '@testing-library/react';
import { MovieDetails } from './MovieDetails';
import { mockDetailedMovie } from '../../test-utils/testData';

describe('MovieDetails', () => {
  it('matches snapshot', () => {
    const { container } = render(<MovieDetails movie={mockDetailedMovie} />);
    expect(container).toMatchSnapshot();
  });
});
