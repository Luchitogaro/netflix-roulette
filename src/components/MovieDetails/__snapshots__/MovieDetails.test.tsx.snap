// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails matches snapshot 1`] = `
<div>
  <div
    class="movie-details"
    data-cy="movie-details"
    data-testid="movie-details"
  >
    <img
      alt="Pulp Fiction"
      class="poster"
      data-cy="movie-poster"
      data-testid="movie-poster"
      src="https://example.com/poster.jpg"
    />
    <div
      class="details"
      data-cy="movie-details-info"
      data-testid="movie-details-info"
    >
      <div
        class="header"
        data-testid="movie-header"
      >
        <h1
          class="title"
          data-cy="movie-title"
          data-testid="movie-title"
        >
          Pulp Fiction
        </h1>
        <span
          class="rating-badge"
          data-cy="movie-rating"
          data-testid="movie-rating"
        >
          8.9
        </span>
      </div>
      <div
        class="genres"
        data-cy="movie-genres"
        data-testid="movie-genres"
      >
        <span
          class="genre-tag"
          data-testid="movie-genre-Crime"
        >
          Crime
        </span>
        <span
          class="genre-tag"
          data-testid="movie-genre-Drama"
        >
          Drama
        </span>
      </div>
      <div
        class="meta"
        data-testid="movie-meta"
      >
        <span
          class="year"
          data-cy="movie-year"
          data-testid="movie-year"
        >
          1994
        </span>
        <span
          class="duration"
          data-cy="movie-duration"
          data-testid="movie-duration"
        >
          2h 34min
        </span>
      </div>
      <p
        class="description"
        data-cy="movie-description"
        data-testid="movie-description"
      >
        A classic crime film directed by Quentin Tarantino.
      </p>
    </div>
  </div>
</div>
`;
